metabase/metabase
# Prevent auto apply filters on slow dashboards

Epic: https://github.com/metabase/metabase/issues/29267
Product doc: https://www.notion.so/metabase/Prevent-auto-apply-filters-in-slow-dashboards-cc02d90dac9e42ee91f34c2a9810212f

This is 2/last of the epic

### Description

Allow users to prevent auto-apply filters on slow dashboard. Otherwise, each filter change could cause a dashboard refresh which could last a long time.

### How to verify

1. Create a dashboard
2. Add a question or model
3. Create a parameter and connect it to the entity added in step 2
4. Modify the filter added in step 3, the dashboard should refresh automatically
5. Open dashboard sidebar and toggle **Auto-apply filters**
6. Modify the filter, and the **Apply** button should appear, the dashboard should not refresh.
7. Click **Apply** button, and the dashboard should refresh.
8. Test the same dashboard in public dashboard, signed embed, and full-app embedding.

### Demo

https://www.loom.com/share/d08afee5fe9340dc9ed385e7c35918f7

### Checklist



- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/metabase/metabase/29569)
